src/debug-generator/debug-generator.c

```diff
@@ -158,8 +163,78 @@ static void install_debug_shell_dropin(void) {
                 log_warning_errno(r, "Failed to write drop-in for debug-shell.service, ignoring: %m");
 }
 
+static int process_unit_credentials(const char *credentials_dir) {
+        _cleanup_close_ int fd = -EBADF;
+        int r;
+
+        assert(credentials_dir);
+
+        fd = open(credentials_dir, O_CLOEXEC|O_DIRECTORY);
```

Comment:
open_credentials_dir()?

Reply:
Doesn't work here, since that doesn't take the credentials directory as an argument. We need to run this on both the regular credentials dir and the encrypted credentials dir.